R/pool-methods.R

```diff
@@ -0,0 +1,89 @@
+#' @include pool.R
+NULL
+
+#' S4 class for compatibility with DBI methods
+#' @export
+setClass("Pool")
+
+## documented manually, with the Pool object
+#' @export
+poolCreate <- function(factory, ...,
+                       minSize = 1, maxSize = Inf,
+                       idleTimeout = 60000, validateTimeout = 2000,
+                       stateEnv = NULL) {
+  pool <- Pool$new(
+    function() {factory(...)},
+    minSize, maxSize,
+    idleTimeout, validateTimeout, stateEnv
+  )
+  ## if a createPool is used with a DBIDriver, make a note
+  ## (to ease dplyr compatibility later on)
+  checkDriver(pool, ...)
```

Comment:
This doesn't make a lot of sense to me (I realize it wasn't introduced in this PR, but I haven't noticed it before). This seems way too tightly bound to DBI to belong in the generic `poolCreate`; can we do this checking in `dbPool`, and store the result in `stateEnv` instead of on an attribute?